### PR TITLE
Fix: python3Packages.tblite by fixing the fortan deps pkgconfig and adding a patch to include multicharge to meson deps

### LINKED ISSUE #223862: tblite-python is not importable / pkgconfig for mtclib, s-dftd3, dftd4
### Describe the bug
tblite is unusable in python at the moment, failing with 
```
Traceback (most recent call last):
  File "/nix/store/pzbm5ci2h0qlnrzkqxhdm9f0vwr5kal6-python3.10-tblite-0.3.0/lib/python3.10/site-packages/tblite/library.py", line 28, in <module>
    from ._libtblite import ffi, lib
ImportError: /nix/store/pzbm5ci2h0qlnrzkqxhdm9f0vwr5kal6-python3.10-tblite-0.3.0/lib/python3.10/site-packages/tblite/_libtblite.cpython-310-x86_64-linux-gnu.so: undefined symbol: __dftd3_model_MOD___vtab_dftd3_model_D3_model

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nix/store/pzbm5ci2h0qlnrzkqxhdm9f0vwr5kal6-python3.10-tblite-0.3.0/lib/python3.10/site-packages/tblite/interface.py", line 26, in <module>
    from . import library
  File "/nix/store/pzbm5ci2h0qlnrzkqxhdm9f0vwr5kal6-python3.10-tblite-0.3.0/lib/python3.10/site-packages/tblite/library.py", line 30, in <module>
    raise ImportError("tblite C extension unimportable, cannot use C-API")
ImportError: tblite C extension unimportable, cannot use C-API
```

I think this is due to incorrect pkgconfig setups - look at 
https://github.com/NixOS/nixpkgs/blob/e608c90a1cf381dde6ac9e0f085337150f2af3e2/pkgs/development/libraries/toml-f/default.nix#L26

which is correct and outputs
```
prefix=/nix/store/8irr24ds1c3r08wclk11gpxp3c7f0p7w-toml-f-0.3.1
libdir=${prefix}/lib
includedir=${prefix}/include

Name: toml-f
Description: A TOML parser implementation for data serialization and deserialization in Fortran
Version: 0.3.1
Libs: -L${libdir} -ltoml-f
Cflags: -I${includedir} -I${includedir}/toml-f/GNU-12.2.0

```
vs

https://github.com/NixOS/nixpkgs/blob/e608c90a1cf381dde6ac9e0f085337150f2af3e2/pkgs/development/libraries/science/chemistry/mctc-lib/default.nix#L27
which is missing the trailing `/`

Which results in a pkgconfig of 
```
prefix=
libdir=${prefix}//lib
includedir=${prefix}//include

Name: mctc-lib
Description: Modular computation tool chain
Requires: 
Version: 0.3.1
Libs: -L${libdir} -lmctc-lib
Cflags: -I${includedir} -I${includedir}/mctc-lib/GNU-12.2.0
```

### Steps To Reproduce
1. Create a nix environment with python3Packages.tblite
2. In the python interpreter, run `import tblite.interface as tb`
3. See failure

### Notify maintainers

@sheepforce 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```

### PR DESCRIPTION
**Pull request by ryanswrt**
###### Description of changes

This fixes #223862 by fixing the pkgconfig paths for a number of fortran libraries, and following the meson build as described in the tblite docs. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/science/chemistry/tblite/python.nix
Comment (sheepforce) [inline pkgs/development/libraries/science/chemistry/tblite/python.nix:47]:
```suggestion
  pythonImportsCheck = [ "tblite" "tblite.interface" ];
```

File: pkgs/development/libraries/science/chemistry/tblite/python.nix
Comment (sheepforce) [inline pkgs/development/libraries/science/chemistry/tblite/python.nix:44]: Could you add a comment what the purpose of this patch is?

**Review by sheepforce — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/science/chemistry/tblite/python.nix
Comment (ryanswrt) [reply pkgs/development/libraries/science/chemistry/tblite/python.nix:46]: Done

**Comment by sheepforce** [comment]
Last point: could you rebase your branch to have one commit per package with standard commit messages?
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

**Comment by ryanswrt** [comment]
> Last point: could you rebase your branch to have one commit per package with standard commit messages? https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

Done, I hope :)

**Comment by sheepforce** [comment]
> Done, I hope :)

Ah, not exactly, I am afraid :grimacing:  Now its one commit for all packages, instead of one per package. Sorry for the trouble :confused: 
Basically something like such a list of commits

  * dftd4: fix pkg-config
  * mctc-lib: fix pkg-config
  * mstore: fix pkg-config
  * multicharge: fix pkg-config
  * simple-dftd3: fix pkg-config
  * tblite: fix python module

If you wish I could also do this, if you allow me to push to your branch in the PR settings, [see here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)

@markuskowa could you merge this when done?

**Comment by ryanswrt** [comment]
> > Done, I hope :)
> 
> Ah, not exactly, I am afraid grimacing Now its one commit for all packages, instead of one per package. Sorry for the trouble confused Basically something like such a list of commits
> 
>     * dftd4: fix pkg-config
> 
>     * mctc-lib: fix pkg-config
> 
>     * mstore: fix pkg-config
> 
>     * multicharge: fix pkg-config
> 
>     * simple-dftd3: fix pkg-config
> 
>     * tblite: fix python module
> 
> 
> If you wish I could also do this, if you allow me to push to your branch in the PR settings, [see here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
> 
> @markuskowa could you merge this when done?

Done - if there's anything else. the "allow edits and access to secrets by maintainers" is checked, not sure if there's another setting to enable edits?

**Comment by markuskowa** [comment]
@GrahamcOfBorg build python3Packages.tblite

**Comment by markuskowa** [comment]
@ryanswrt can you please the change the commit messages into nixpkgs standard format:
`<package name>: fix pkg-config` (like @sheepforce wrote it in his comment above)

**Comment by ryanswrt** [comment]
> @ryanswrt can you please the change the commit messages into nixpkgs standard format: `<package name>: fix pkg-config` (like @sheepforce wrote it in his comment above)

Done

**Review by sheepforce — APPROVED** [review]

**Comment by sheepforce** [comment]
Glorious, thank you very much @ryanswrt 
LGTM.

**Pull request merged**
